x/gamm/pool-models/balancer/codec.go

```diff
@@ -20,8 +21,13 @@ func RegisterLegacyAminoCodec(cdc *codec.LegacyAmino) {
 
 func RegisterInterfaces(registry codectypes.InterfaceRegistry) {
 	registry.RegisterInterface(
-		"osmosis.gamm.v1beta1.PoolI",
-		(*types.PoolI)(nil),
+		"osmosis.swaprouter.v1beta1.PoolI",
+		(*swaproutertypes.PoolI)(nil),
+		&Pool{},
+	)
+	registry.RegisterInterface(
+		"osmosis.gamm.v1beta1.CFMMPoolI",
+		(*types.CFMMPoolI)(nil),
```

Comment:
imo we need to keep the old names here, otherwise we would need a relatively complex state migration.

Reply:
Alternative is we make state migration code that takes the old type and casts it to a new type. This seems like a complex integrator break, that doesn't seem worth it imo

Reply:
Something to keep in mind on this matter - we have an e2e test that pre-creates pools before the upgrade, runs some swaps against them, performs the upgrade, and runs some swaps again.

If the e2e test is passing, do you think still that the migrations would be necessary?

Reply:
Either way, I will try to understand whether this is unsafe by looking into `RegisterInterface(...)` logic

Reply:
Reverted to old name to avoid breaking clients:

[9f01762](https://github.com/osmosis-labs/osmosis/pull/3560/commits/9f01762419a4c8941bd637579e59e41ab0892d1f)